client/src/components/JitsiGame.js

```diff
@@ -93,34 +111,57 @@ class JitsiGame {
    * generates main jitsi game class with provided data client configuration
    * @param {object} config
    */
-    constructor(config) {
+    constructor(config, Game) {
+        // Need this._roomname
+        this.Game = Game;
         this.config = config;
         console.log('constructing now');
         this._dataClient = new DataClient(this.config);
         this._api = false;
+        this._roomName = false;
     }
 
     /**
    * Generates a new game window
    * @param {string} selector
    */
-    newGame(selector) {
-        return new Promise((resolve, reject) => {
-            this._api.executeCommand('hangup');
-            this._api.dispose();
-            const randomRoomName = generateRoomWithoutSeparator();
+    newGame(selector, roomName) {
+        this._api.executeCommand('hangup');
+        this._api.dispose();
+        document.querySelector('#gamelist').innerHTML = ' ';
+        this.Game.renderGame();
+        this.Game.handleClickEvents();
```

Comment:
Why not call handleClickEvents inside renderGame?  That way you don't have to duplicate the same logic below.

Reply:
Makes total sense to me thanks for catching that Aaron. I've made the change in the newest commit.